src/operator/nn/mkldnn/mkldnn_pooling.cc

```diff
@@ -255,7 +252,10 @@ void MKLDNNPoolingCompute(const OpContext &ctx, const PoolingParam &param,
                           const NDArray &in_data, const OpReqType req,
                           const NDArray &out_data, const NDArray *workspace) {
   auto fwd = GetPoolingFwd(param, ctx.is_train, in_data, out_data);
-  fwd.SetDataHandle(in_data, out_data, workspace);
+  auto input_mem = in_data.GetMKLDNNData();
+  auto out_mem_t = CreateMKLDNNMem(out_data, fwd.fwd_pd_->dst_primitive_desc(), req, &in_data);
```

Comment:
do we need write_inplace here?

Reply:
do you mean we should remove the CreateMKLDNNMem / CommitOutput here as we don't need to worry about writing inplace?

Reply:
i mean you don't need to pass `&in_data`.